app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -216,6 +216,18 @@ protected void enterRootModule() {
         startActivityForResult(i, GET_COMMAND);
     }
 
+    private boolean useGridMenu(String menuId) {
+        if(menuId == null) {
+            menuId = org.commcare.suite.model.Menu.ROOT_MENU_ID;
+        }
+        if(DeveloperPreferences.isGridMenuEnabled()) { return true; }
+        String commonDisplayStyle = platform.getMenuDisplayStyle(menuId);
+        if("grid".equals(commonDisplayStyle)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Refactored for static conditional. I left the static MENU_STYLE_GRID in this file because the logic for what styles are available/used is platform-dependent and no other object has the right level of context (It would be weird to put it in /commcare/ because commcare doesn't actually support the grid).